invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/UppyUploader/UppyUploader.js

```diff
@@ -0,0 +1,312 @@
+  proudlyDisplayPoweredByUppy: false,
+  hideProgressAfterFinish: true,
+  hidePauseResumeButton: true,
+  disableLocalFiles: false,
+  // Allows to select both files and folders
+  fileManagerSelectionType: "both",
+  height: "100%",
+  width: "100%",
+  autoOpen: null,
+  autoOpenFileEditor: false,
+};
+
+export const UppyUploaderComponent = ({
+  config,
+  files,
+  isDraftRecord,
+  hasParentRecord,
+  quota,
+  permissions,
+  record,
+  initializeFileUpload,
+  finalizeUpload,
+  deleteFile,
+  uploadPart,
+  saveAndFetchDraft,
+  setUploadProgress,
+  importParentFiles,
+  importButtonIcon,
+  importButtonText,
+  isFileImportInProgress,
+  fileUploadConcurrency,
+  decimalSizeDisplay,
+  filesLocked,
+  allowEmptyFiles,
+  ...uiProps
+}) => {
+  // We extract the working copy of the draft stored as `values` in formik
+  const { values: formikDraft } = useFormikContext();
+  const { filesList } = useFilesList(files);
+  const locale = useUppyLocale();
+  const filesEnabled = _get(formikDraft, "files.enabled", false);
+  const filesSize = filesList.reduce((totalSize, file) => (totalSize += file.size), 0);
+  const lockFileUploader = !isDraftRecord && filesLocked;
+  const filesLeft = filesList.length < quota.maxFiles;
+  const displayImportBtn =
+    filesEnabled && isDraftRecord && hasParentRecord && !filesList.length;
+
+  const restrictions = React.useMemo(
+    () => ({
+      minFileSize: allowEmptyFiles ? 0 : 1,
+      maxNumberOfFiles: quota.maxFiles - filesList.length,
+      maxTotalFileSize: quota.maxStorage - filesSize,
+    }),
+    [allowEmptyFiles, quota, filesList, filesSize]
+  );
+
+  const [uppy] = useState(() =>
+    new Uppy({
+      debug: false,
+      autoProceed: false,
+      restrictions,
+      locale,
+    })
+      .use(InvenioMultipartUploader, {
+        limit: fileUploadConcurrency,
+        // Bind Redux file actions to the uploader plugin
+        initializeFileUpload,
+        finalizeUpload,
+        saveAndFetchDraft,
+        setUploadProgress,
+        uploadPart,
+        abortUpload: (file) => deleteFile(file),
+        checkPartIntegrity: true,
+      })
+      .use(ImageEditor)
+  );
+
+  React.useEffect(() => {
+    const uploaderPlugin = uppy.getPlugin("InvenioMultipartUploader");
+    if (uploaderPlugin) {
+      // Synchronize uploader state with current formik state
+      uploaderPlugin.draftRecord = formikDraft;
+    }
+  }, [uppy, formikDraft]);
+
+  React.useEffect(() => {
+    // Synchronize uppy locale with i18next
+    uppy.setOptions({ locale });
+  }, [uppy, locale]);
+
+  React.useEffect(() => {
+    uppy.setOptions({ restrictions });
+  }, [uppy, restrictions]);
+
+  React.useEffect(() => {
+    const dashboardPlugin = uppy.getPlugin("uppy-uploader-dashboard");
+    if (!dashboardPlugin) {
+      return;
+    }
+    dashboardPlugin.setOptions({ disabled: !filesLeft });
+  }, [uppy, filesLeft]);
+
+  return (
+    <Grid className="file-uploader">
```

Comment:
nit: ideally we miss here some overridable on the container part as it is done below for the `ImportButton`